index.js

```diff
@@ -472,8 +472,20 @@ function writeArrayLen(marker, value) {
 }
 
 function readCompressedData(buffer, state) {
-  const compressedData = buffer.slice(state.pos + 4, buffer.indexOf(COMPRESSED_DATA_END, state.pos));
-  return zlib.unzipSync(compressedData, { finishFlush: zlib.Z_SYNC_FLUSH });
+  const data = buffer.slice(state.pos + 4,
+                            buffer.indexOf(COMPRESSED_DATA_END, state.pos) + COMPRESSED_DATA_END.length);
+
+  // drop 4 bytes away after every chunk
+  const chunkSize = 64 * 1024;
+  const chunks = [];
+  let pos = 0;
+  while (pos < data.length) {
+      chunks.push(data.slice(pos, pos + chunkSize));
+      pos += chunkSize + 4;
+  }
+  const compressedData = Buffer.concat(chunks);
+
+  return zlib.deflateSync(compressedData, { finishFlush: zlib.Z_SYNC_FLUSH });
```

Comment:
This should be unzipSync

Reply:
`unzipSync` autodetects the format and in this case the compression algorithm is deflate. Or is there some other reason to use `unzipSync`?
https://nodejs.org/api/zlib.html#zlib_class_zlib_unzip